knowledge_graphs/performance_monitor.py

```
@@ -0,0 +1,497 @@
+        """Monitor a Neo4j query execution.
+        
+        Args:
+            query_name: Name/identifier for the query
+            query: Cypher query string
+            params: Query parameters
+        """
+        start_time = time.time()
+        
+        # Hash query for caching analysis
+        import hashlib
+        query_hash = hashlib.md5(query.encode()).hexdigest()[:8]
```

Ruff (0.12.2)

Probable use of insecure hash functions in `hashlib`: `md5`

(S324)